src/BenchmarkDotNet/EventHandlers/IEventHandler.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using BenchmarkDotNet.Toolchains.Results;
+using BenchmarkDotNet.Validators;
+
+namespace BenchmarkDotNet.EventHandlers
+{
+    public interface IEventHandler
+    {
+        void HandleStartValidationStage();
+        void HandleUnsupportedBenchmark(BenchmarkCase benchmarkCase);
+        void HandleValidationError(ValidationError validationError);
+        void HandleStartBuildStage();
+        void HandleBuildFailed(BenchmarkCase benchmarkCase, BuildResult buildResult);
+        void HandleStartRunStage();
+        void HandleRunBenchmarksInType(Type type, IReadOnlyList<BenchmarkCase> benchmarks);
+        void HandleCompletedBenchmarksInType(Type type, Summary summary);
```

Comment:
the naming is inconsistent: Start&Stop vs Run&Completed. We should choose one schema and stick to it.